aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -332,6 +407,261 @@ def _build_middleware_stack(self, router_middlewares: List[Callable[..., Any]])
 
         self._middleware_stack_built = True
 
+    @property
+    def dependant(self) -> "Dependant":
+        if self._dependant is None:
+            from aws_lambda_powertools.event_handler.openapi.dependant import get_dependant
+
+            self._dependant = get_dependant(path=self.path, call=self.func)
+
+        return self._dependant
+
+    @property
+    def body_field(self) -> Optional["ModelField"]:
+        if self._body_field is None:
+            from aws_lambda_powertools.event_handler.openapi.dependant import get_body_field
+
+            self._body_field = get_body_field(dependant=self.dependant, name=self.operation_id)
+
+        return self._body_field
+
+    def _get_openapi_path(
+        self,
+        *,
+        dependant: "Dependant",
+        operation_ids: Set[str],
+        model_name_map: Dict["TypeModelOrEnum", str],
+        field_mapping: Dict[Tuple["ModelField", Literal["validation", "serialization"]], "JsonSchemaValue"],
+    ) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+        from aws_lambda_powertools.event_handler.openapi.dependant import get_flat_params
+
+        path = {}
+        definitions: Dict[str, Any] = {}
+
+        # Gather all the route parameters
+        operation = self._openapi_operation_metadata(operation_ids=operation_ids)
+        parameters: List[Dict[str, Any]] = []
+        all_route_params = get_flat_params(dependant)
+        operation_params = self._openapi_operation_parameters(
+            all_route_params=all_route_params,
+            model_name_map=model_name_map,
+            field_mapping=field_mapping,
+        )
+        parameters.extend(operation_params)
+
+        # Add the parameters to the OpenAPI operation
+        if parameters:
+            all_parameters = {(param["in"], param["name"]): param for param in parameters}
+            required_parameters = {(param["in"], param["name"]): param for param in parameters if param.get("required")}
+            all_parameters.update(required_parameters)
+            operation["parameters"] = list(all_parameters.values())
+
+        # Add the request body to the OpenAPI operation, if applicable
+        if self.method.upper() in METHODS_WITH_BODY:
+            request_body_oai = self._openapi_operation_request_body(
+                body_field=self.body_field,
+                model_name_map=model_name_map,
+                field_mapping=field_mapping,
+            )
+            if request_body_oai:
+                operation["requestBody"] = request_body_oai
+
+        # Add the response to the OpenAPI operation
+        responses = operation.setdefault("responses", {})
+        success_response = responses.setdefault("200", {})
+        success_response["description"] = self.response_description or _DEFAULT_OPENAPI_RESPONSE_DESCRIPTION
+        success_response["content"] = {"application/json": {"schema": {}}}
+        json_response = success_response["content"].setdefault("application/json", {})
```

Comment:
Is it right? It seems like I always just set HTTP 200 and don't consider replacing the description for HTTP 200 or adding a new response status code. I tried `@app.get("/hello", responses={"302": {"description": "a"}})` and `@app.get("/hello", responses={"200": {"description": "a"}})` and didn't work both. Always just return the schema with HTTP 200 and 422.

Reply:
You are right. I fixed that and created tests.

By default, if you don't specify `responses`, we'll add 200 and 422. You can customize the 200 response description by passing `response_description`. If you instead pass `responses`, we will not touch it and use it directly in the schema.